### PR TITLE
Add docs for dev and ci/cd workflows

### LINKED ISSUE #3074: Complete skaffold workflows documentation
Issue to track finishing the docs for skaffold workflows

### PR DESCRIPTION
**Pull request by nkubala**
Fixes #3074

### CODE REVIEW

File: docs/content/en/docs/workflows/dev.md [hunk 1]
```diff
@@ -4,4 +4,44 @@ linkTitle: "Continuous development"
 weight: 1
 ---
 
-{{< todo 3074>}} dev page
\ No newline at end of file
+## `skaffold dev`
+
+`skaffold dev` enables continuous local development on an application.
+While in `dev` mode, Skaffold will watch an application's source files, and when it detects changes,
```
Comment (tejal29) [inline docs/content/en/docs/workflows/dev.md]:
```suggestion
While in `dev` mode, Skaffold will watch an application's source files, and when it detects changes, it
```

File: docs/content/en/docs/workflows/dev.md [hunk 2]
```diff
@@ -4,4 +4,44 @@ linkTitle: "Continuous development"
...
+will rebuild your images (or sync files to your running containers), push any new images, and redeploy the application to your cluster.
+
+`skaffold dev` is considered Skaffold's main mode of operation, as it allows you
+to leverage all of the main features of Skaffold in a continuous way while iterating
```
Comment (tejal29) [inline docs/content/en/docs/workflows/dev.md]:
```suggestion
to leverage all of the rich features of Skaffold like [`file-sync`](), `port-forward` in a continuous way while iterating
```

File: docs/content/en/docs/workflows/dev.md [hunk 3]
```diff
@@ -4,4 +4,44 @@ linkTitle: "Continuous development"
...
+on your application.
+
+
+## Dev loop
+
+When `skaffold dev` is run, Skaffold will first do a full build and deploy of all artifacts specified in the `skaffold.yaml`, identical behavior to `skaffold run`. Upon successful build and deploy, Skaffold will start watching all source file dependencies for all artifacts specified in the project. As changes are made to these source files, Skaffold will rebuild the associated artifacts, and redeploy the new changes to your cluster.
```
Comment (tejal29) [inline docs/content/en/docs/workflows/dev.md]: I dont think we have `skaffold run` documented any where. we can drop it or link to existing CLI doc on run.

Comment (tejal29) [inline docs/content/en/docs/workflows/dev.md]:
```suggestion
When `skaffold dev` is run, Skaffold will first do a full build and deploy of all artifacts specified in the `skaffold.yaml`, similar to `skaffold run`. Upon successful build and deploy, Skaffold will start watching all source file dependencies for all artifacts specified in the project. As changes are made to these source files, Skaffold will rebuild the associated artifacts, and redeploy the new changes to your cluster. 
You can control this behaviour <point to trigger types docs>
```

File: docs/content/en/docs/workflows/dev.md [hunk 4]
```diff
@@ -4,4 +4,44 @@ linkTitle: "Continuous development"
...
+
+The dev loop will run until the user cancels the Skaffold process with `Ctrl+C`. Upon receiving this signal, Skaffold will clean up all deployed artifacts on the active cluster, meaning that Skaffold won't abandon any Kubernetes resources that it created throughout the lifecycle of the run.
```
Comment (tejal29) [inline docs/content/en/docs/workflows/dev.md]:
```suggestion
The dev loop will run until the user cancels the Skaffold process with `Ctrl+C`. Upon receiving this signal, Skaffold will clean up all deployed artifacts on the active cluster.
```
We should also list we clean artifacts built by the dev loop which can be controlled via `--no-prune` and `--no-prune-children` flag

File: docs/content/en/docs/workflows/dev.md [hunk 5]
```diff
@@ -4,4 +4,44 @@ linkTitle: "Continuous development"
...
+
+
+## Precedence of Actions
+
+The actions performed by Skaffold during the dev loop have precendence over one another, so that behavior is always predictable. The order of actions is:
+
+1) File Sync
```
Comment (tejal29) [inline docs/content/en/docs/workflows/dev.md]: WHATTTTT
I thought it was build, deploy, filesync.

Comment (tejal29) [reply docs/content/en/docs/workflows/dev.md]: we use `1.` everywhere else.

File: docs/content/en/docs/workflows/dev.md [hunk 6]
```diff
@@ -4,4 +4,44 @@ linkTitle: "Continuous development"
...
+
+
+## File Watcher and Watch Modes
+
+Skaffold computes the dependencies for each artifact based on the builder being used, and the root directory of the artifact. Once all source file dependencies are computed, in `dev` mode, Skaffold will continuously watch these files for changes in the background, and conditionally re-run the loop when changes are detected.
+
+By default, Skaffold uses `fsnotify` to monitor events on the local filesystem. Skaffold also supports a `polling` mode where the filesystem is checked for changes on a configurable interval, or a `manual` mode, where Skaffold waits for user input to check for file changes. These watch modes can be configured through the `--trigger` flag.
+
+
+## Control API
```
Comment (tejal29) [inline docs/content/en/docs/workflows/dev.md]: I think we should explain trigger types here and then control api.

Comment (balopat) [reply docs/content/en/docs/workflows/dev.md]: also, I'm writing a section about the Control API :)

File: docs/content/en/docs/workflows/ci-cd.md [hunk 7]
```diff
@@ -4,4 +4,75 @@ linkTitle: "CI/CD with Skaffold"
 weight: 3
```
Comment (tejal29) [inline docs/content/en/docs/workflows/ci-cd.md]: Add a diff of 10.

File: docs/content/en/docs/workflows/dev.md (hunk 6 above)
Comment (nkubala) [reply docs/content/en/docs/workflows/dev.md]: I just wanted to have a little section about it here for discoverability, and then link in the full docs to the control API. I mentioned the trigger types above this section. do you think I should remove this?

File: docs/content/en/docs/workflows/dev.md (hunk 1 above)
Comment (nkubala) [reply docs/content/en/docs/workflows/dev.md]: I actually think this flows a bit better without the `it`:

Skaffold **will** do something, and when something happens, **will** do something else.

File: docs/content/en/docs/workflows/dev.md (hunk 6 above)
Comment (balopat) [reply docs/content/en/docs/workflows/dev.md]: I'd remove it, it is an advanced topic for those who want to integrate tools with `skaffold dev` - it is not essential for the CLI users. WDYT?

File: docs/content/en/docs/workflows/ci-cd.md [hunk 8]
```diff
@@ -1,7 +1,88 @@
...
+```code
+skaffold render | kubectl apply -f -
+```
+
+gives us the one line output telling us the only thing we need to know:
+```code
+pod/getting-started configured
+```
+
+## Waiting for Skaffold deployments using `healthcheck`
```
Comment (balopat) [inline docs/content/en/docs/workflows/ci-cd.md]: are we going to write this separately?

Comment (tejal29) [reply docs/content/en/docs/workflows/ci-cd.md]: yes. i am going to write this

File: docs/content/en/docs/workflows/dev.md (hunk 6 above)
Comment (tejal29) [reply docs/content/en/docs/workflows/dev.md]: I agree, control API is not for CLI user and more of integration for tools.

Comment (nkubala) [reply docs/content/en/docs/workflows/dev.md]: sounds reasonable, I do think CLI users could still potentially make use of the control API, but it's not a first class feature. I cut down the blurb to just explain what it is and why you might use it, and then pointed to the API docs

File: docs/content/en/docs/workflows/dev.md [hunk 9]
```diff
@@ -4,4 +4,38 @@ linkTitle: "Continuous development"
...
+
+By default, the dev loop will carry out all actions (as needed) each time a file is changed locally, with the exception of operating in `manual` trigger mode. However, individual actions can be gated off by user input through the Skaffold API.
+
+With this API, users can tell Skaffold to wait for user input before performing any of these actions, even if the requisite files were changed on the filesystem. By doing so, users can "queue up" changes while they are iterating locally, and then have Skaffold rebuild and redeploy only when asked. This can be very useful when builds are happening more frequently than desired, when builds or deploys take a long time or are otherwise very costly, or when users want to integrate other tools with `skaffold dev`.
+
+For more documentation, see the Skaffold API Docs. (//TODO(balintp) write docs)
```
Comment (balopat) [inline docs/content/en/docs/workflows/dev.md]:
```suggestion
For more documentation, see the [Skaffold API Docs]({{<relref "/docs/design/api" >}})
```

**Comment by balopat** [comment]
:shipit: :ship:

**Pull request merged**